materialize-postgres/driver.go

```diff
@@ -187,6 +192,8 @@ func (c client) FetchSpecAndVersion(ctx context.Context, specs sql.Table, materi
 }
 
 func (c client) ExecStatements(ctx context.Context, statements []string) error {
+	// Execute statements within a transaction.
+	statements = append(append([]string{"BEGIN;"}, statements...), "END;")
```

Comment:
This will have an effect on all other statements being run as well as the table creation SQL, so I think if we want to limit the blast radius we can move this `BEGIN` and `END` additions to driver.go where we are creating the table creation SQL, what do you think?